### PR TITLE
feat: support openharmony platform

### PR DESCRIPTION
**Pull request by richerfu**
[OpenHarmony](https://gitee.com/openharmony) is a new operating system, currently mainly used for mobile devices. The triplet for rust build is `aarch64-unknown-linux-ohos`. Although it is a linux-like system, we cannot obtain root permissions. Therefore, our implementation is similar to Android based with `FD`.

### CODE REVIEW

File: examples/dev-config.rs [hunk 1]
```diff
@@ -41,7 +41,7 @@ fn main_entry(quit: Receiver<()>) -> Result<(), BoxError> {
         .destination((10, 0, 0, 1))
         .up();
 
-    #[cfg(target_os = "linux")]
+    #[cfg(all(target_os = "linux", not(target_env = "ohos")))]
```
[the remark this answers is not in the document]
Comment (richerfu) [reply examples/dev-config.rs:44]: Yes, but we can't use linux implementation directly, we cannot obtain root permissions

[the remark this answers is not in the document]
Comment (ssrlive) [reply examples/dev-config.rs:44]: So, please remove this, and add some dummy functions to `ohos` implementation.

**Comment by ssrlive** [comment]
Can you provide a test suit on Github workflow?

**Comment by ssrlive** [comment]
Can you provide a test suit on `Github` workflow for `openharmony`?

**Comment by richerfu** [comment]
> Can you provide a test suit on `Github` workflow for `openharmony`?

Just lint and compiler is ok ?

**Comment by ssrlive** [comment]
Yes

**Pull request merged**